src/utils.test.js

```diff
@@ -573,6 +578,176 @@ describe('isFormEditing', () => {
+
+    const result = getNextSelectedRowsState(selectedRows, row);
+
+    expect(result).toEqual({});
+  });
+
+  it('should not mutate the original selectedRows object', () => {
+    const row = { id: 1, name: 'John Doe' };
+    const selectedRows = { [row.id]: row };
+
+    const result = getNextSelectedRowsState(selectedRows, row);
+
+    // Ensure original selectedRows object is not mutated
+    expect(selectedRows).toEqual({ [row.id]: row });
+    expect(result).not.toBe(selectedRows);
+  });
+});
+describe('isPrintable', () => {
+  test('returns true when pickSlips contain a match for requestId', () => {
+    const requestId = '123';
+    const pickSlips = [
+      { request: { requestID: '123' } },
+      { request: { requestID: '456' } },
+    ];
+
+    const result = isPrintable(requestId, pickSlips);
+
+    expect(result).toBe(true);
+  });
+
+  test('returns false when pickSlips do not contain a match for requestId', () => {
+    const requestId = '789';
+    const pickSlips = [
+      { request: { requestID: '123' } },
+      { request: { requestID: '456' } },
+    ];
+
+    const result = isPrintable(requestId, pickSlips);
+
+    expect(result).toBe(false);
+  });
+
+  test('returns false when pickSlips is undefined', () => {
+    const requestId = '123';
+    const pickSlips = undefined;
+
+    const result = isPrintable(requestId, pickSlips);
+
+    expect(result).toBe(false);
+  });
+});
+
+describe('getSelectedSlipData', () => {
+  const pickSlipsData = [
+    { 'request.requestID': 1, data: 'some data 1' },
+    { 'request.requestID': 2, data: 'some data 2' },
+    { 'request.requestID': 3, data: 'some data 3' },
+  ];
+
+  it('should return an empty array if selectedRequestId is not found', () => {
+    const selectedRequestId = 999;
+    const result = getSelectedSlipData(pickSlipsData, selectedRequestId);
+    expect(result).toEqual([]);
+  });
+
+  it('should return an array with the selected slip data if found', () => {
+    const selectedRequestId = 2;
+    const result = getSelectedSlipData(pickSlipsData, selectedRequestId);
+    const expectedResult = [{ 'request.requestID': 2, data: 'some data 2' }];
+    expect(result).toEqual(expectedResult);
+  });
+
+  it('should return an array even if only one matching slip is found', () => {
+    const selectedRequestId = 3;
+    const result = getSelectedSlipData(pickSlipsData, selectedRequestId);
+    const expectedResult = [{ 'request.requestID': 3, data: 'some data 3' }];
+    expect(result).toEqual(expectedResult);
+  });
+
+  it('should handle an empty pickSlipsData array', () => {
+    const selectedRequestId = 1;
+    const result = getSelectedSlipData([], selectedRequestId);
+    expect(result).toEqual([]);
+  });
+});
+
+describe('selectedRowsNonPrintable', () => {
+  test('should return true when selectedRows is empty', () => {
+    const pickSlipsData = [
+      { 'request.requestID': '1' },
+      { 'request.requestID': '2' },
```

Comment:
Could you please use one naming approach through tests?
For example in some places you use `it` in other places you use `test`. We usually use `it`, lets use this approach. 
```suggestion
  it('should handle an empty pickSlipsData array', () => {
    const selectedRequestId = 1;
    const result = getSelectedSlipData([], selectedRequestId);
    expect(result).toEqual([]);
  });
});

describe('selectedRowsNonPrintable', () => {
  it('should return true when selectedRows is empty', () => {
    const pickSlipsData = [
      { 'request.requestID': '1' },
      { 'request.requestID': '2' },
```